### ISSUE TITLE
Required + sometimes not working as expected

### ISSUE THREAD

**Issue by j-jalving** [spatie/statamic-responsive-images#228]
### Bug description

The responsive images field is always required, even when it's invisible using a "Show when" condition and required + sometimes validation is set

### How to reproduce

1. Add a "Show when" condition to the responsive image field
2. Enable the Required and Sometimes validators.

The field is now always required, even when it's not visible.

### Environment

```yaml
Environment
Application Name: Website
Laravel Version: 10.11.0
PHP Version: 8.2.4
Composer Version: 2.5.5
Environment: local
Debug Mode: ENABLED
URL: website.test
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: ses
Queue: sync
Session: file

Localization
Installed: en
LaravelLang\Attributes\Plugin: v2.3.3
LaravelLang\HttpStatuses\Plugin: v3.3.1
LaravelLang\Lang\Plugin: 12.21.5
Protected Locales: en
Publisher Version: v14.6.4

Statamic
Addons: 7
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.1.2 Solo

Statamic Addons
aerni/livewire-forms: 8.0.1
aerni/social-links: 3.1.0
aryehraber/statamic-logbook: 3.1.0
dryven/faviconator: 1.2.0
jonassiewertsen/statamic-livewire: 2.12.0
spatie/statamic-responsive-images: 4.0.2
statamic/seo-pro:
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

runtime (new)

**Comment by ncla** [comment]
Should be fixed in v4.0.3. Do let me know if it's working.

**Comment by generator85** [comment]
Seems to work now, thanks!
